scripts/docker/test.bash

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -eu
+set -o pipefail
+
+function test() {
+  local package="${1:?Provide a package}"
+  local sub_package="${2:-}"
+
+  export DIR=${package}
+  . <(/ci/shared/helpers/extract-default-params-for-task.bash /ci/shared/tasks/run-bin-test/linux.yml)
+
+  export DEFAULT_PARAMS="ci/routing-release/default-params/run-bin-test/linux.yml"
+  export ENVS="DB=${DB}"
+  export GOFLAGS="-buildvcs=false"
+  /ci/shared/tasks/run-bin-test/task.bash "${sub_package}"
+}
+
+if [[ -n "${1:-}" ]]; then
+  test "src/code.cloudfoundry.org/${1}" "${2:-}"
+else
+  for component in gorouter cf-tcp-router multierror route-registrar routing-api routing-api-cli routing-info; do
```

Comment:
Yes, that's a good catch, we should be pulling in the list from `index.yml` instead of the list here.

Reply:
Started looking into fixing this today, but i'm not sure I like the idea of requiring the CI repo to be present in order to run local tests on routing-release. 

We could have it default to auto-generating in routing-release if the CI repo isn't present to find the index.yml file, but then risk a false sense of security between local testing not matching CI testing.

Thinking about auditing, we would want to catch the following:
- we tested fewer components than we expected
- we tested more components than we expected

These would require two separate lists, and each would need to be updated as a component is added. I'm not sure if this is beneficial since if no action is done after adding a new component, it isn't tested, and nothing is brought to our attention. If it is added to one list but not the other, we get alerted to add it to another list. Most likely it would get added to the list of repos to test first, and then alerted to add to the audit list. 

When using testing uses the same hardcoded list used for auditing, the audit becomes:
- the list of components that were tested matches the codebases in the repo that we wanted to be tested

Like you said in the other PR's comments, it's difficult to programatically decide what components are important, since some are submodules from other releases, or different paths entirely (./src/code.cloudfoundry vs ./src). 

It might be easier to update other releases to ensure everything needing to be tested is in src/code.cloudfoundry.org, and don't include components there if they shouldn't be tested (using a tools package to pull them in as vendored dependencies).

Did you have other methodologies for auditing what's being tested that I'm not coming up with?